crates/criticalup-core/src/download_server_client.rs

```diff
@@ -374,17 +422,46 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn cache_is_constructed() {
+    async fn cache_content_is_migrated() {
         let test_env = TestEnvironment::with().download_server().prepare().await;
+        // creating a file at the incorrect path
+        let cache_dir = test_env.config().paths.cache_dir.clone();
+        let old_path = cache_dir
+            .join("artifacts")
+            .join("ferrocene")
+            .join("stable-25.05.0");
 
+        let foo = old_path.join("foo.txt");
+
+        fs::create_dir_all(old_path.clone()).await.unwrap();
+        fs::File::create_new(foo).await.unwrap();
+
+        test_env
+            .download_server()
+            // the tested function that migrates the cache
+            .product_release_cache_path("ferrocene", "stable-25.05.0")
+            .await;
+        let expected = "artifacts/products/ferrocene/releases/stable-25.05.0";
+        // the file must be found in the new cache
+        let new_path = cache_dir.join(expected).join("foo.txt");
+
+        assert!(new_path.exists());
```

Comment:
To make test intention explicit we should assert:
- **before execution**,  of the new functionality that the old path exists and the new doesn't
after execution:  assert that the oldld path doesn't exist anymore, the new one does (this one is in the code already)

Reply:
Done in 771459b